include/acquire.zarr.h

```diff
@@ -0,0 +1,274 @@
+#ifndef H_ACQUIRE_ZARR_V0
+#define H_ACQUIRE_ZARR_V0
+
+#include "zarr.types.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    typedef struct ZarrStreamSettings_s ZarrStreamSettings;
+    typedef struct ZarrStream_s ZarrStream;
+
+    /**
+     * @brief Get the version of the Zarr API.
+     * @return The version of the Zarr API.
+     */
+    uint32_t Zarr_get_api_version();
+
+    /**
+     * @brief Set the log level for the Zarr API.
+     * @param level The log level.
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus Zarr_set_log_level(ZarrLogLevel level);
+
+    /**
+     * @brief Get the log level for the Zarr API.
+     * @return The log level for the Zarr API.
+     */
+    ZarrLogLevel Zarr_get_log_level();
+
+    /**
+     * @brief Get the message for the given status code.
+     * @param status The status code.
+     * @return A human-readable status message.
+     */
+    const char* Zarr_get_error_message(ZarrStatus status);
+
+    /**
+     * @brief Create a Zarr stream settings struct.
+     * @return A pointer to the Zarr stream settings struct, or NULL on failure.
+     */
+    ZarrStreamSettings* ZarrStreamSettings_create();
+
+    /**
+     * @brief Destroy a Zarr stream settings struct.
+     * @details This function frees the memory allocated for the Zarr stream
+     * settings struct.
+     * @param[in] settings The Zarr stream settings struct.
+     */
+    void ZarrStreamSettings_destroy(ZarrStreamSettings* settings);
+
+    /**
+     * @brief Copy a Zarr stream settings struct.
+     * @param[in] settings The Zarr stream settings struct to copy.
+     * @return A copy of the Zarr stream settings struct.
+     */
+    ZarrStreamSettings* ZarrStreamSettings_copy(
+      const ZarrStreamSettings* settings);
+
+    /**
+     * @brief Set store path and S3 settings for the Zarr stream.
+     * @param[in, out] settings
+     * @param[in] store_path The store path for the Zarr stream. Directory path
+     * when acquiring to the filesystem, key prefix when acquiring to S3.
+     * @param[in] bytes_of_store_path The length of @p store_path in bytes,
+     * including the null terminator.
+     * @param[in] s3_settings Optional S3 settings. If NULL, the store path is
+     * assumed to be a directory path.
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus ZarrStreamSettings_set_store(ZarrStreamSettings* settings,
```

Comment:
Since there are two distinct store types (filesystem and S3), I recommend offering separate initialization functions for each type. This approach not only simplifies the API but also makes the intended usage clearer to the user.

Having dedicated functions allows for more specific parameter requirements and error handling tailored to each store type, which can enhance the overall user experience. It reduces ambiguity and helps prevent misuse, ultimately leading to a more intuitive and maintainable interface.

Another way to enhance the API would be to separate the store parameters object from the settings object.